spades.wdl

```diff
@@ -0,0 +1,79 @@
+task spades {
+    String outputDir
+    String? preCommand
+    File? read1
+    File? read2
+    File? singleRead
+    File? interlacedReads
+    File? sangerReads
+    File? pacbioReads
+    File? nanoporeReads
+    File? tslrContigs
+    File? trustedContigs
+    File? untrustedContigs
+    Boolean? singleCell
+    Boolean? metagenomic
+    Boolean? rna
+    Boolean? plasmid
+    Boolean? ionTorrent
+    Boolean? onlyErrorCorrection
+    Boolean? onlyAssembler
+    Boolean? careful
+    Boolean? disableGzipOutput
+    Boolean? disableRepeatResolution
+    File? dataset
+    Int threads
+    Int memoryGb
+    File? tmpDir
+    String? k
+    Float? covCutoff
+    Int? phredOffset
+
+    command {
+        set -e -o pipefail
+        ${preCommand}
+        spades.py \
+        ${"-o " + outputDir} \
+        ${true="--sc" false="" singleCell} \
+        ${true="--meta" false="" metagenomic} \
+        ${true="--rna" false="" rna} \
+        ${true="--plasmid" false="" plasmid} \
+        ${true="--iontorrent" false="" ionTorrent} \
+        ${"--12 " + interlacedReads } \
+        ${"-1 " + read1 } \
+        ${"-2 " + read2 } \
+        ${"-s " + singleRead } \
```

Comment:
If this is replaced with `${true="-1" false="-s" defined(read2)} read1`, read1 can be made mandatory. If I'm not mistaken.

Reply:
:+1: